homework1/client/src/game.js

```diff
@@ -0,0 +1,37 @@
+// default interval as 1 second
+import constants from './constants';
+
+const interval = 1000;
+
+/**
+ * loop is main loop of the game, which will be executed once every second (
+ * based on the interval variable configuration)
+ */
+export function loop (store) {
+	// TODO: increment counter based on the generators in the state
+	// hint: read how many "generators" in store and iterate through them to
+	//       count how many value to increment to "resource"
+	
+	if (window.globalGeneratorRate > 0) {
+		if (window.store.incrementTime)
+			clearTimeout(window.store.incrementTime);
+		store.dispatch({
+					type: constants.actions.INCREMENT,
+					payload: window.globalGeneratorRate
```

Comment:
While you are not wrong here using a global state. It's _anti-pattern_ to use state under global state. This becomes hard for you to transition into later assignment when we want to separate away the _meta data_ from the display.